### DISCUSSION TITLE
Naming conventions for single interface implementation

### DISCUSSION THREAD

**Discussion by ndr-brt** [eclipse-edc/Connector#553] [Ideas]
Hi, I found two different conventions on naming the single implementations of an interface.

e.g.
`DataProxyManagerImpl` for `DataProxyManager`
and
`DefaultProxyEntryHandlerRegistry` for `ProxyEntryHandlerRegistry`

Would be good to choose a single convention and document it.
Personally I like the `Default*` one.

**Comment by paullatzelsperger** [comment]
In this particular case you're right, as the `DataProxyManager` is actually a bit of a misnomer - it should be named `DataProxyRegistry` IMO. 

However, on a wider scale: to me there is a slight difference between a `Default` and an `Impl`. The way I see it, the `Default*` convention should eventually supersede the `InMemory*` stuff. So for example by default, for every registry there is an im-mem implementation (which it essentially a map) which is called `Default...Registry`. It is conceivable, that someone wants to implement a registry based on some storage other than memory. 
The `Impl` should be used for something that really only ever has one implementation, such as the `BlobStoreApiImpl` or the `DataFlowManagerImpl`.

**Reply by ndr-brt** [reply]
That's fair, so we could define a standard where:
- single interface implementation is called `<interface name>Impl`
- on multiple implementations if there's an implementation that's used by default but it could be swapped by other implementations
